sfxclient/gometrics.go

```diff
@@ -12,14 +12,34 @@ var startTime = time.Now()
 // GoMetricsSource is a singleton Collector that collects basic go system stats.  It currently
 // collects from runtime.ReadMemStats and adds a few extra metrics like uptime of the process
 // and other runtime package functions.
-var GoMetricsSource Collector = &goMetrics{}
+var GoMetricsSource MemCallback = &goMetrics{}
```

Comment:
i kind of have a gut feeling but can't put my finger on it other than that's what other libs do. but perhaps use a Get() and Set() for the MemCallback?  See opentracing.